x-pack/plugin/inference/src/yamlRestTest/resources/rest-api-spec/test/inference/20_semantic_text_field_mapper.yml

```diff
@@ -0,0 +1,153 @@
+setup:
+  - skip:
+      version: " - 8.12.99"
+      reason: semantic_text introduced in 8.13.0 # TODO change when 8.13.0 is released
+
+  - do:
+      inference.put_model:
+        task_type: sparse_embedding
+        inference_id: sparse-inference-id
+        body: >
+          {
+            "service": "test_service",
+            "service_settings": {
+              "model": "my_model",
+              "api_key": "abc64"
+            },
+            "task_settings": {
+            }
+          }
+  - do:
+      inference.put_model:
+        task_type: text_embedding
+        inference_id: dense-inference-id
+        body: >
+          {
+            "service": "text_embedding_test_service",
+            "service_settings": {
+              "model": "my_model",
+              "dimensions": 10,
+              "api_key": "abc64"
+            },
+            "task_settings": {
+            }
+          }
+
+  - do:
+      indices.create:
+        index: test-index
+        body:
+          mappings:
+            properties:
+              sparse_field:
+                type: semantic_text
+                model_id: sparse-inference-id
+              dense_field:
+                type: semantic_text
+                model_id: dense-inference-id
+              non_inference_field:
+                type: text
+
+---
+"Sparse vector results format":
+    - do:
+        index:
+          index: test-index
+          id: doc_1
+          body:
+            non_inference_field: "you know, for testing"
+            _semantic_text_inference:
+              sparse_field:
+                model_settings:
+                  inference_id: sparse-inference-id
+                  task_type: sparse_embedding
+                inference_results:
+                  - text: "inference test"
+                    inference:
+                      feature_1: 0.1
+                      feature_2: 0.2
+                      feature_3: 0.3
+                      feature_4: 0.4
+                  - text: "another inference test"
+                    inference:
+                      feature_1: 0.1
+                      feature_2: 0.2
+                      feature_3: 0.3
+                      feature_4: 0.4
```

Comment:
We should check the response too? And then verify that the mapping is as expected?

Reply:
I misread, I thought it was a bulk index but it's just one document sorry.